supervisor/tests/test_dispatchers.py

```diff
@@ -512,6 +512,76 @@ def test_syslog_logfile_deprecated(self):
             self._makeOne(process)
             self.assertEqual(len(w), 1)
 
+    def test_no_logfile_and_no_syslog(self):
+        from supervisor.datatypes import boolean, logfile_name
```

Comment:
No other existing tests in this file import things from `supervisor.datatypes` although they do similar configuration setup.

Reply:
Other tests don't read special LOGFILE_ datatypes (datatypes.py:91-92) into their configs. I used the datatypes functions so that the test config input (e.g. lines 521-22) is exactly what would be read from a real config file. I believe this makes the tests more readable and more robust.